### ISSUE TITLE
"ValueError: fetch requires an index" on empty VCF file with existing tabix index

### ISSUE THREAD

**Issue by eldariont** [pysam-developers/pysam#904]
Hi,

I encountered `ValueError: fetch requires an index` when opening an empty VCF file although the tabix index existed. In my opinion, fetching from an empty VCF file should not throw an exception but just return an empty set of variants.

To reproduce this (pysam version 0.15.4 from pip), I attach my empty [VCF file](https://github.com/pysam-developers/pysam/files/4350002/empty.vcf.gz). To create a tabix index: `tabix -p vcf empty.vcf.gz`

Then, in python:
```
from pysam import VariantFile
vcf = VariantFile("empty.vcf.gz")
vcf.fetch("chr21", 100, 1000)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pysam/libcbcf.pyx", line 4382, in pysam.libcbcf.VariantFile.fetch
ValueError: fetch requires an index
```

Cheers
David

**Comment by jmarshall** [comment]
Thanks for the bug report and test case. I agree, `fetch requires an index` is not accurate here: there **is** an index; this is different from the case when there is no _.tbi_ file alongside the _.vcf.gz_.

This is because of a very clever `__bool__()` routine that somewhat arbitrarily considers an index with no reference sequences to be False:
https://github.com/pysam-developers/pysam/blob/beb6bc9eafc0978ee341a712f544167bd6eafc07/pysam/libcbcf.pyx#L3593-L3594

However rather than change that we can change the test leading to this exception to test that an index has been opened instead. (This is the only such `if [not] self.index` statement on this kind of index.) Fixed in beb6bc9eafc0978ee341a712f544167bd6eafc07.

**Comment by eldariont** [comment]
Great, thanks for fixing this so quickly! 👍
